pkg/types/ccipocr3/plugin_commit_types.go

```diff
@@ -1,10 +1,27 @@
 package ccipocr3
 
+// CommitPluginReport contains the necessary information to commit CCIP
+// messages from potentially many source chains, to a single destination chain.
+//
+// It must consist of either:
+//
+//  1. a non-empty MerkleRoots array, or
+//  2. a non-empty PriceUpdates array
+//
+// If neither of the above is provided the report is considered empty and should
+// not be transmitted on-chain.
+//
+// In the event the MerkleRoots array is non-empty, it may also contain
+// RMNSignatures, if RMN is configured for some lanes involved in the commitment.
+// A report with RMN signatures but without merkle roots is invalid.
 type CommitPluginReport struct {
-	MerkleRoots  []MerkleRootChain `json:"merkleRoots"`
-	PriceUpdates PriceUpdates      `json:"priceUpdates"`
+	MerkleRoots   []MerkleRootChain   `json:"merkleRoots"`
+	PriceUpdates  PriceUpdates        `json:"priceUpdates"`
+	RMNSignatures []RMNECDSASignature `json:"rmnSignatures"`
 }
 
+// Deprecated: don't use this constructor, just create a CommitPluginReport struct directly.
+// Will be removed in a future version once all uses have been replaced.
 func NewCommitPluginReport(merkleRoots []MerkleRootChain, tokenPriceUpdates []TokenPrice, gasPriceUpdate []GasPriceChain) CommitPluginReport {
```

Comment:
This function is being deprecated to avoid breaking changes, adding a param would mean a breaking change and a desperate bump across chainlink-ccip and chainlink